uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt311/client/sdkmqtt/Mqtt311ConnectionImpl.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt311.client.sdkmqtt;
+
+import com.aws.greengrass.testing.mqtt5.client.GRPCClient;
+import com.aws.greengrass.testing.mqtt5.client.MqttConnection;
+import com.aws.greengrass.testing.mqtt5.client.MqttLib;
+import com.aws.greengrass.testing.mqtt5.client.exceptions.MqttException;
+import lombok.NonNull;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.IMqttToken;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implementation of MQTT 3.1.1 connection based on AWS IoT SDK.
+ */
+public class Mqtt311ConnectionImpl implements MqttConnection {
+    private static final Logger logger = LogManager.getLogger(Mqtt311ConnectionImpl.class);
+    private static final String EXCEPTION_WHEN_CONNECTING = "Exception occurred during connect";
+
+    private final AtomicBoolean isClosing = new AtomicBoolean();
+    private final AtomicBoolean isConnected = new AtomicBoolean();
+
+    private final GRPCClient grpcClient;
+    private final IMqttClient mqttClient;
+    private int connectionId = 0;
+
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();        // TODO: use DI
+
+    /**
+     * Creates a MQTT 3.1.1 connection.
+     *
+     * @param connectionParams the connection parameters
+     * @param grpcClient the consumer of received messages and disconnect events
+     * @throws MqttException on errors
+     */
+    public Mqtt311ConnectionImpl(@NonNull MqttLib.ConnectionParams connectionParams, @NonNull GRPCClient grpcClient)
+            throws org.eclipse.paho.client.mqttv3.MqttException {
+        super();
+        this.grpcClient = grpcClient;
+        this.mqttClient = createClient(connectionParams);
+    }
+
+    @Override
+    public ConnectResult start(long timeout, int connectionId) throws MqttException {
+        this.connectionId = connectionId;
+        try {
+            IMqttToken token = mqttClient.connectWithResult(new MqttConnectOptions());
```

Comment:
Out of scope current task. 

For future tasks:
1. timeout is not used, please use async interfaces if sync has no time limits
2. empty new MqttConnectOptions()) - probably it is reason when IoT core reject unauthorized connection